packages/asset-swapper/src/utils/market_operation_utils/balancer_utils.ts

```diff
@@ -0,0 +1,76 @@
+import { BigNumber } from '@0x/utils';
+import { bmath, getPoolsWithTokens, parsePoolData } from '@balancer-labs/sor';
+import * as _ from 'lodash';
+
+export interface BalancerPool {
+    id: string;
+    balanceIn: BigNumber;
+    balanceOut: BigNumber;
+    weightIn: BigNumber;
+    weightOut: BigNumber;
+    swapFee: BigNumber;
+    spotPrice?: BigNumber;
+    slippage?: BigNumber;
+    limitAmount?: BigNumber;
+}
+interface CacheValue {
+    timestamp: number;
+    pools: BalancerPool[];
+}
+const THIRTY_MINUTES_MS = 30 * 60 * 1000; // tslint:disable-line:custom-no-magic-numbers
```

Comment:
IMO 30 mins is fine for a cache of existing pools. I.e a new pool added takes 30m till it is served by AssetSwapper.

30 mins is too long for rates within the pool and determining the route based on these values. Especially with sub 1% slippage.

I'd suggest we cache for at most a minute and move towards a model where the pools are discovered (cached) but queried on-demand.

Reply:
Agreed. I wonder if it'd be worth differentiating pool discovery vs pool state. idk if there would be a substantial difference in latency between querying the state of the cached pool addresses versus calling `getPoolsWithTokens`

Reply:
yeah I assume TheGraph is fairly quick, which is fine for pool discovery. 

But ultimately I think we'd want to get the rates as live as possible to reduce our potential for revert.